packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1267,6 +1330,9 @@ class SelectionOverlay {
       _isDraggingEndHandle = false;
       return;
     }
+    if (_blockEndHandleDrag) {
```

Comment:
I wonder if you could get rid of the _blockStart/EndHandleDrag booleans by checking something like this here instead:

```dart
_isDraggingStartHandle && defaultTargetPlatform == TargetPlatform.iOS || defaultTargetPlatform == TargetPlatform.macOS || kIsWeb
```

That could be a getter to make things cleaner.